### PR TITLE
Changes to send data only once per 24 hours

### PR DESCRIPTION
**Pull request by SnchitGrover**
Logic to log events only once per (24 hours or session), whatever comes first

### CODE REVIEW

**Comment by SnchitGrover** [comment]
@niteskum @shubhsnov @swmitra @narayani28 please review changes.

**Comment by shubhsnov** [comment]
Hey @SnchitGrover Can you go ahead and get yourself added to the Adobe org as mentioned [here](https://git.corp.adobe.com/OpenSourceAdvisoryBoard/handbook/blob/master/GitHub-Adobe-Org-Management.md#request-access-to-our-adobe-github-org).

The CLA check won't pass until you do that. One you get added, close and reopen the pull request.

File: src/extensions/default/HealthData/HealthDataManager.js [hunk 1]
```diff
@@ -336,6 +330,17 @@ define(function (require, exports, module) {
...
+    /**
+     * This function is auto called after 24 hours to empty the map
+     * Map is used to make sure that we send an event only once per 24 hours
+     **/
+
+    function emptyAnalyticsMap() {
+        HealthLogger.analyticsEventMap.clear();
+        setTimeout(emptyAnalyticsMap, ONE_DAY);
+    }
+    setTimeout(emptyAnalyticsMap, ONE_DAY);
```
Comment (shubhsnov) [inline src/extensions/default/HealthData/HealthDataManager.js]: We can use setInterval once?

File: src/extensions/default/HealthData/HealthDataManager.js [hunk 2]
```diff
@@ -336,6 +330,17 @@ define(function (require, exports, module) {
         return result.promise();
     }
 
+    /**
+     * This function is auto called after 24 hours to empty the map
+     * Map is used to make sure that we send an event only once per 24 hours
+     **/
+
+    function emptyAnalyticsMap() {
+        HealthLogger.analyticsEventMap.clear();
```
Comment (shubhsnov) [inline src/extensions/default/HealthData/HealthDataManager.js]: Shouldn't we just clear it once per day. Once cleared we shouldn't clear it for any other sessions as well.
Same for sending the data.
This wouldn't work across sessions right?

File: src/extensions/default/HealthData/HealthDataManager.js [hunk 3]
```diff
@@ -308,22 +308,16 @@ define(function (require, exports, module) {
...
-        var options = {
-            location: {
-                scope: "default"
-            }
-        };
-
         if (isHDTracking) {
-            isEventDataAlreadySent = PreferencesManager.getViewState(Eventparams.eventName);
-            PreferencesManager.setViewState(Eventparams.eventName, 1, options);
+            isEventDataAlreadySent = HealthLogger.analyticsEventMap.get(Eventparams.eventName);
```
Comment (shubhsnov) [inline src/extensions/default/HealthData/HealthDataManager.js]: Is` HealthLogger.analyticsEventMap` persistent across sessions?

**Review by shubhsnov — APPROVED** [review]

**Comment by shubhsnov** [comment]
Since we're going ahead with non-persistent data, this looks okay to me. LGTM 👍

**Comment by shubhsnov** [comment]
@SnchitGrover pull in the laster master once and do a merge.

**Comment by niteskum** [comment]
Looks like **/*global Map*/**  missed in HealthLogger.js
add this in HealthLogger.js

**Pull request merged**